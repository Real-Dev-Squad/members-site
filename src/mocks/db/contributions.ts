export const allContributionsData = [
  {
    prList: [
      {
        title: "application API ( do not review or merge)",
        state: "open",
        createdAt: "2023-08-09T16:59:09Z",
        updatedAt: "2023-08-09T17:09:39Z",
        url: "https://github.com/Real-Dev-Squad/website-backend/pull/1391",
        raisedBy: "vinayak-trivedi",
      },
    ],
    task: {},
  },
  {
    prList: [
      {
        title: "Social link fix",
        state: "closed",
        createdAt: "2023-07-23T03:06:17Z",
        updatedAt: "2023-07-23T06:01:13Z",
        url: "https://github.com/Real-Dev-Squad/members-site/pull/63",
        raisedBy: "vinayak-trivedi",
      },
    ],
    task: {},
  },
];
