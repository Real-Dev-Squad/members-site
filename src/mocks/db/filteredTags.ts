export const filteredTags = [
  {
    levelId: "1dOI6j3YNW4XQR5rwQsm",
    levelName: "7",
    levelValue: 7,
    name: "EMBER level 7",
    tagId: "4qvOozqaWIiHT4fBlVjk",
    tagName: "EMBER",
    tagType: "SKILL",
  },
  {
    levelId: "G7GMnx0EiE1kBIoLrjTs",
    levelName: "6",
    levelValue: 6,
    name: "EMBER level 6",
    tagId: "4qvOozqaWIiHT4fBlVjk",
    tagName: "EMBER",
    tagType: "SKILL",
  },
  {
     levelId: "RaG21J0vRxGSPApT2Eqa",
     levelName: "5",
     levelValue: 5,
     name: "EMBER level 5",
     tagId: "4qvOozqaWIiHT4fBlVjk",
     tagName: "EMBER",
     tagType: "SKILL",
   },
  {
     levelId: "Y6BmnuF9SstgHHw475SU",
     levelName: "1",
     levelValue: 1,
     name: "EMBER level 1",
     tagId: "4qvOozqaWIiHT4fBlVjk",
     tagName: "EMBER",
     tagType: "SKILL",
   },
  {
     levelId: "ZPqw5Qpv7IAiJ1DfoXj7",
     levelName: "4",
     levelValue: 4,
     name: "EMBER level 4",
     tagId: "4qvOozqaWIiHT4fBlVjk",
     tagName: "EMBER",
     tagType: "SKILL",
   },
  {
     levelId: "i6H3KVymvblRUN1RUxLY",
     levelName: "2",
     levelValue: 2,
     name: "EMBER level 2",
     tagId: "4qvOozqaWIiHT4fBlVjk",
     tagName: "EMBER",
     tagType: "SKILL",
   },
  {
     levelId: "lqaCbeSZ0gohOhijp0Vx",
     levelName: "8",
     levelValue: 8,
     name: "EMBER level 8",
     tagId: "4qvOozqaWIiHT4fBlVjk",
     tagName: "EMBER",
     tagType: "SKILL",
   },
  {
     levelId: "1dOI6j3YNW4XQR5rwQsm",
     levelName: "7",
     levelValue: 7,
     name: "TEST-SKILL-3 level 7",
     tagId: "TldcNbsU8vMPpjlkXTvf",
     tagName: "TEST-SKILL-3",
     tagType: "SKILL",
   },
  {
     levelId: "G7GMnx0EiE1kBIoLrjTs",
     levelName: "6",
     levelValue: 6,
     name: "TEST-SKILL-3 level 6",
     tagId: "TldcNbsU8vMPpjlkXTvf",
     tagName: "TEST-SKILL-3",
     tagType: "SKILL",
   },
  {
     levelId: "GgF7zx2i8vfekX7pRkeg",
     levelName: "3",
     levelValue: 3,
     name: "TEST-SKILL-3 level 3",
     tagId: "TldcNbsU8vMPpjlkXTvf",
     tagName: "TEST-SKILL-3",
     tagType: "SKILL",
   },
  {
     levelId: "RaG21J0vRxGSPApT2Eqa",
     levelName: "5",
     levelValue: 5,
     name: "TEST-SKILL-3 level 5",
     tagId: "TldcNbsU8vMPpjlkXTvf",
     tagName: "TEST-SKILL-3",
     tagType: "SKILL",
   },
  {
     levelId: "Y6BmnuF9SstgHHw475SU",
     levelName: "1",
     levelValue: 1,
     name: "TEST-SKILL-3 level 1",
     tagId: "TldcNbsU8vMPpjlkXTvf",
     tagName: "TEST-SKILL-3",
     tagType: "SKILL",
   },
  {
     levelId: "ZPqw5Qpv7IAiJ1DfoXj7",
     levelName: "4",
     levelValue: 4,
     name: "TEST-SKILL-3 level 4",
     tagId: "TldcNbsU8vMPpjlkXTvf",
     tagName: "TEST-SKILL-3",
     tagType: "SKILL",
   },
  {
     levelId: "i6H3KVymvblRUN1RUxLY",
     levelName: "2",
     levelValue: 2,
     name: "TEST-SKILL-3 level 2",
     tagId: "TldcNbsU8vMPpjlkXTvf",
     tagName: "TEST-SKILL-3",
     tagType: "SKILL",
   },
  {
     levelId: "lqaCbeSZ0gohOhijp0Vx",
     levelName: "8",
     levelValue: 8,
     name: "TEST-SKILL-3 level 8",
     tagId: "TldcNbsU8vMPpjlkXTvf",
     tagName: "TEST-SKILL-3",
     tagType: "SKILL",
   },
  {
     levelId: "G7GMnx0EiE1kBIoLrjTs",
     levelName: "6",
     levelValue: 6,
     name: "TEST-SKILL-2 level 6",
     tagId: "aNQEOkM8JZyLibCfAzTC",
     tagName: "TEST-SKILL-2",
     tagType: "SKILL",
   },
  {
     levelId: "GgF7zx2i8vfekX7pRkeg",
     levelName: "3",
     levelValue: 3,
     name: "TEST-SKILL-2 level 3",
     tagId: "aNQEOkM8JZyLibCfAzTC",
     tagName: "TEST-SKILL-2",
     tagType: "SKILL",
   },
  {
     levelId: "RaG21J0vRxGSPApT2Eqa",
     levelName: "5",
     levelValue: 5,
     name: "TEST-SKILL-2 level 5",
     tagId: "aNQEOkM8JZyLibCfAzTC",
     tagName: "TEST-SKILL-2",
     tagType: "SKILL",
   },
  {
     levelId: "Y6BmnuF9SstgHHw475SU",
     levelName: "1",
     levelValue: 1,
     name: "TEST-SKILL-2 level 1",
     tagId: "aNQEOkM8JZyLibCfAzTC",
     tagName: "TEST-SKILL-2",
     tagType: "SKILL",
   },
  {
     levelId: "ZPqw5Qpv7IAiJ1DfoXj7",
     levelName: "4",
     levelValue: 4,
     name: "TEST-SKILL-2 level 4",
     tagId: "aNQEOkM8JZyLibCfAzTC",
     tagName: "TEST-SKILL-2",
     tagType: "SKILL",
   },
  {
     levelId: "i6H3KVymvblRUN1RUxLY",
     levelName: "2",
     levelValue: 2,
     name: "TEST-SKILL-2 level 2",
     tagId: "aNQEOkM8JZyLibCfAzTC",
     tagName: "TEST-SKILL-2",
     tagType: "SKILL",
   },
  {
     levelId: "lqaCbeSZ0gohOhijp0Vx",
     levelName: "8",
     levelValue: 8,
     name: "TEST-SKILL-2 level 8",
     tagId: "aNQEOkM8JZyLibCfAzTC",
     tagName: "TEST-SKILL-2",
     tagType: "SKILL",
   },
  {
     levelId: "1dOI6j3YNW4XQR5rwQsm",
     levelName: "7",
     levelValue: 7,
     name: "DEBUGGING level 7",
     tagId: "syTMXXjWBGtWGhFPVZ1H",
     tagName: "DEBUGGING",
     tagType: "SKILL",
   },
  {
     levelId: "G7GMnx0EiE1kBIoLrjTs",
     levelName: "6",
     levelValue: 6,
     name: "DEBUGGING level 6",
     tagId: "syTMXXjWBGtWGhFPVZ1H",
     tagName: "DEBUGGING",
     tagType: "SKILL",
   },
  {
     levelId: "GgF7zx2i8vfekX7pRkeg",
     levelName: "3",
     levelValue: 3,
     name: "DEBUGGING level 3",
     tagId: "syTMXXjWBGtWGhFPVZ1H",
     tagName: "DEBUGGING",
     tagType: "SKILL",
   },
  {
     levelId: "RaG21J0vRxGSPApT2Eqa",
     levelName: "5",
     levelValue: 5,
     name: "DEBUGGING level 5",
     tagId: "syTMXXjWBGtWGhFPVZ1H",
     tagName: "DEBUGGING",
     tagType: "SKILL",
   },
  {
     levelId: "Y6BmnuF9SstgHHw475SU",
     levelName: "1",
     levelValue: 1,
     name: "DEBUGGING level 1",
     tagId: "syTMXXjWBGtWGhFPVZ1H",
     tagName: "DEBUGGING",
     tagType: "SKILL",
   },
  {
     levelId: "ZPqw5Qpv7IAiJ1DfoXj7",
     levelName: "4",
     levelValue: 4,
     name: "DEBUGGING level 4",
     tagId: "syTMXXjWBGtWGhFPVZ1H",
     tagName: "DEBUGGING",
     tagType: "SKILL",
   },
  {
     levelId: "i6H3KVymvblRUN1RUxLY",
     levelName: "2",
     levelValue: 2,
     name: "DEBUGGING level 2",
     tagId: "syTMXXjWBGtWGhFPVZ1H",
     tagName: "DEBUGGING",
     tagType: "SKILL",
   },
  {
     levelId: "lqaCbeSZ0gohOhijp0Vx",
     levelName: "8",
     levelValue: 8,
     name: "DEBUGGING level 8",
     tagId: "syTMXXjWBGtWGhFPVZ1H",
     tagName: "DEBUGGING",
     tagType: "SKILL",
   },
  {
     levelId: "1dOI6j3YNW4XQR5rwQsm",
     levelName: "7",
     levelValue: 7,
     name: "TEST-SKILL-1 level 7",
     tagId: "yd38KmQItUVdJw0woPoe",
     tagName: "TEST-SKILL-1",
     tagType: "SKILL",
   },
  {
     levelId: "G7GMnx0EiE1kBIoLrjTs",
     levelName: "6",
     levelValue: 6,
     name: "TEST-SKILL-1 level 6",
     tagId: "yd38KmQItUVdJw0woPoe",
     tagName: "TEST-SKILL-1",
     tagType: "SKILL",
   },
  {
     levelId: "GgF7zx2i8vfekX7pRkeg",
     levelName: "3",
     levelValue: 3,
     name: "TEST-SKILL-1 level 3",
     tagId: "yd38KmQItUVdJw0woPoe",
     tagName: "TEST-SKILL-1",
     tagType: "SKILL",
   },
  {
     levelId: "RaG21J0vRxGSPApT2Eqa",
     levelName: "5",
     levelValue: 5,
     name: "TEST-SKILL-1 level 5",
     tagId: "yd38KmQItUVdJw0woPoe",
     tagName: "TEST-SKILL-1",
     tagType: "SKILL",
   },
  {
     levelId: "Y6BmnuF9SstgHHw475SU",
     levelName: "1",
     levelValue: 1,
     name: "TEST-SKILL-1 level 1",
     tagId: "yd38KmQItUVdJw0woPoe",
     tagName: "TEST-SKILL-1",
     tagType: "SKILL",
   },
  {
     levelId: "ZPqw5Qpv7IAiJ1DfoXj7",
     levelName: "4",
     levelValue: 4,
     name: "TEST-SKILL-1 level 4",
     tagId: "yd38KmQItUVdJw0woPoe",
     tagName: "TEST-SKILL-1",
     tagType: "SKILL",
   },
  {
     levelId: "i6H3KVymvblRUN1RUxLY",
     levelName: "2",
     levelValue: 2,
     name: "TEST-SKILL-1 level 2",
     tagId: "yd38KmQItUVdJw0woPoe",
     tagName: "TEST-SKILL-1",
     tagType: "SKILL",
   },
  {
     levelId: "lqaCbeSZ0gohOhijp0Vx",
     levelName: "8",
     levelValue: 8,
     name: "TEST-SKILL-1 level 8",
     tagId: "yd38KmQItUVdJw0woPoe",
     tagName: "TEST-SKILL-1",
     tagType: "SKILL",
   },
];
