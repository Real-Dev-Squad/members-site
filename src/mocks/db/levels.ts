export const levelsData = [
  {
    id: 'ORS84peu3fAyPi54CNDt',
    date: {
      _seconds: 1670107075,
      _nanoseconds: 971000000,
    },
    createdBy: 'XAF7rSUvk4p0d098qWYS',
    name: '1',
    value: 1,
  },
  {
    id: 'QEQfB3MhcZCB2nnGzj9T',
    date: {
      _seconds: 1670516361,
      _nanoseconds: 177000000,
    },
    createdBy: 'XAF7rSUvk4p0d098qWYS',
    name: '2',
    value: 2,
  },
  {
    id: 'ZbxSl5h3AgFx4G1Axb5P',
    date: {
      _seconds: 1670105941,
      _nanoseconds: 299000000,
    },
    createdBy: 'XAF7rSUvk4p0d098qWYS',
    name: '3',
    value: 3,
  },
  {
    id: 'cDdQXU0GKxVBIRDnBKpH',
    date: {
      _seconds: 1670516153,
      _nanoseconds: 348000000,
    },
    createdBy: 'XAF7rSUvk4p0d098qWYS',
    name: '4',
    value: 4,
  },
  {
    id: 'mtipLVmqbvsJGNT2cjiJ',
    date: {
      _seconds: 1677194133,
      _nanoseconds: 987000000,
    },
    createdBy: 'XAF7rSUvk4p0d098qWYS',
    name: '5',
    value: 5,
  },
  {
    id: 'pcfFBfsmICeOkipFmHTP',
    date: {
      _seconds: 1670518024,
      _nanoseconds: 296000000,
    },
    createdBy: 'XAF7rSUvk4p0d098qWYS',
    name: '6',
    value: 6,
  },
  {
    id: 'qHW1Dtte0DtfqzlMOzqP',
    date: {
      _seconds: 1677194139,
      _nanoseconds: 304000000,
    },
    createdBy: 'XAF7rSUvk4p0d098qWYS',
    name: '7',
    value: 7,
  },
  {
    id: 'w9k8hqCVvDzYpOn7qaGs',
    date: {
      _seconds: 1670729417,
      _nanoseconds: 638000000,
    },
    createdBy: 'XAF7rSUvk4p0d098qWYS',
    name: '8',
    value: 8,
  },
];
