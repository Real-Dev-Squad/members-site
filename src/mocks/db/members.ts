export const membersData = [
  {
    id: "2LYhIJRQBPypkcCuuW7q",
    incompleteUserDetails: true,
    github_display_name: "Manish Devrani",
    updated_at: 1691432824942,
    roles: {
      archived: false,
      in_discord: false,
      member: false,
      super_user: false,
    },
    github_id: "manish591",
    isMember: false,
  },
  {
    id: "3XDExvejTrg6YbeSPKtr",
    incompleteUserDetails: false,
    github_display_name: null,
    roles: {
      archived: false,
      in_discord: false,
      member: false,
    },
    last_name: "trivedi",
    github_id: "vinayak-trivedi",
    first_name: "vinayak",
    username: "vinayak",
    isMember: false,
  },
  {
    id: "3kcyMhE0PyEjLrxKjLcu",
    incompleteUserDetails: true,
    discordJoinedAt: "2022-01-08T19:01:40.148000+00:00",
    discordId: "482066749106290690",
    github_display_name: "Pallab Sonowal",
    roles: {
      archived: false,
      in_discord: true,
      member: false,
    },
    github_id: "pallabez",
    isMember: false,
  },
];
