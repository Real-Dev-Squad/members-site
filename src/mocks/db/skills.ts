export const skillsData = [
  {
    id: 'BV5CmTdrxkkQoPwxnPOy',
    itemId: 'KTkF4vAd6tsuBw84oZXt',
    itemType: 'USER',
    levelId: 'GgF7zx2i8vfekX7pRkeg',
    levelName: '3',
    levelValue: 3,
    tagId: '4qvOozqaWIiHT4fBlVjk',
    tagName: 'EMBER',
    tagType: 'SKILL',
  },
  {
    id: 'HONaBLoC6d9rbugt9EK1',
    itemId: 'KTkF4vAd6tsuBw84oZXt',
    itemType: 'USER',
    levelId: '1dOI6j3YNW4XQR5rwQsm',
    levelName: '7',
    levelValue: 7,
    tagId: 'aNQEOkM8JZyLibCfAzTC',
    tagName: 'TEST-SKILL-2',
    tagType: 'SKILL',
  },
];
