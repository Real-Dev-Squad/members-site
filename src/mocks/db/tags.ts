export const tagsData = [
  {
    id: '19Dow91hLL8mIKZWzu2s',
    date: {
      _seconds: 1682483283,
      _nanoseconds: 958000000,
    },
    reason: 'adding skills to users',
    createdBy: 'XAF7rSUvk4p0d098qWYS',
    name: 'EMBER',
    type: 'SKILL',
  },
  {
    id: 'S02fv4NPNDucOBHsMoji',
    date: {
      _seconds: 1670515455,
      _nanoseconds: 617000000,
    },
    reason: 'adding skills to users',
    createdBy: 'XAF7rSUvk4p0d098qWYS',
    type: 'SKILL',
    name: 'TEST-SKILL-2',
  },
  {
    id: 'UNr4wG3uKxCsD1GORUzi',
    date: {
      _seconds: 1670105813,
      _nanoseconds: 422000000,
    },
    reason: 'adding skills to users',
    createdBy: 'XAF7rSUvk4p0d098qWYS',
    type: 'SKILL',
    name: 'TEST-SKILL-1',
  },
  {
    id: 'fzCcp4slrqIENsks4Tai',
    date: {
      _seconds: 1677194119,
      _nanoseconds: 484000000,
    },
    reason: 'adding skills to users',
    createdBy: 'XAF7rSUvk4p0d098qWYS',
    name: 'DEBUGGING',
    type: 'SKILL',
  },
  {
    id: 'nf7GTgWI1bT1C0VUQp9O',
    date: {
      _seconds: 1670518002,
      _nanoseconds: 287000000,
    },
    reason: 'adding skills to users',
    createdBy: 'XAF7rSUvk4p0d098qWYS',
    type: 'SKILL',
    name: 'TEST-SKILL-3',
  },
];
