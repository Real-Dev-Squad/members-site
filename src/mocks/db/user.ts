export const userData = {
  id: '3XDExvejTrg6YbeSPKtr',
  incompleteUserDetails: false,
  github_display_name: null,
  roles: {
    archived: false,
    in_discord: false,
    member: false,
  },
  last_name: 'trivedi',
  github_id: 'vinayak-trivedi',
  first_name: 'vinayak',
  username: 'vinayak',
};

export const selfUser = {
  id: 'Ui8cQ2AJHV3LlNsDTvQX',
  incompleteUserDetails: true,
  github_display_name: 'Anish Pawaskar',
  github_id: 'anishpawaskar',
  roles: {
    archived: false,
    in_discord: false,
    super_user: false,
  },
  username: 'anish',
};
