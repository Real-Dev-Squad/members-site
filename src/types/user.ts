import { UserType } from "../../src/components/MembersSectionNew/types/MembersSection.type";

type PictureType = {
  publicId: string;
  url: string;
};

type RolesType = {
  archived: boolean;
  member: boolean;
};

export type UsersResponseType = {
  message: string;
  users: UserType[];
};