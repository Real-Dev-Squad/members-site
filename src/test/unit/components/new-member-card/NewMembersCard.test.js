import { renderWithProviders } from '../../../../test__utils/renderWithProvides';
import React from 'react';
import NewMembersCard from '../../../../components/NewMemberSection/NewMemberCard/index';
import { userData } from '../../../../mocks/db/user';

jest.mock('next/router', () => ({
  useRouter() {
    return {
      route: '/',
      pathname: '',
      query: '',
      asPath: '',
    };
  },
}));

describe('NewMembersCard', () => {
  it('renders the component', () => {
    const user = userData;

    const { container } = renderWithProviders(<NewMembersCard user={user} />);
    expect(container).toMatchSnapshot();
  });
});
