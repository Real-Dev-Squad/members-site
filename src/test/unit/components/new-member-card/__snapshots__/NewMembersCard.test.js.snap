// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewMembersCard renders the component 1`] = `
<div>
  <div
    class="flex bg-white max-w-sm rounded-md transition-shadow duration-300 h-35 overflow-hidden"
  >
    <div
      class="p-4 rounded-md"
    >
      <div
        class="text-center p-2"
      >
        <img
          alt="Sunny Kumar"
          class="rounded-full mx-auto"
          data-nimg="1"
          decoding="async"
          height="60"
          loading="lazy"
          src="/_next/image?url=%2F..%2F..%2F..%2Fpublic%2Fimages%2FAvatar.png&w=128&q=75"
          srcset="/_next/image?url=%2F..%2F..%2F..%2Fpublic%2Fimages%2FAvatar.png&w=64&q=75 1x, /_next/image?url=%2F..%2F..%2F..%2Fpublic%2Fimages%2FAvatar.png&w=128&q=75 2x"
          style="color: transparent;"
          width="60"
        />
        <p
          class="py-2 font-bold"
        >
          Sunny Kumar
        </p>
      </div>
    </div>
  </div>
</div>
`;
